Filter/AbstractDateFilter.php

```diff
@@ -132,18 +123,8 @@ public function getDefaultOptions()
      */
     public function getRenderSettings()
     {
-        $name = 'sonata_type_filter_date';
-
-        if ($this->time) {
-            $name .= 'time';
-        }
-
-        if ($this->range) {
-            $name .= '_range';
-        }
-
```

Comment:
You should use the `switch` I proposed and throw a deprecation here when appropriate, and still honor the old values.